fix(offlinemsg): make faux offline messages purely event based

anthonybilinski opened the pull request:
Offline messages are now sent as soon as we see our friend come online, and at no other time. Fixes 2 minute wait time before attempting to send if message is entered while you or friend is offline, removes 2 minute constant retry timer, removes 250ms delay between seeing friend come online and sending offline messages.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/qtox/qtox/4975)
<!-- Reviewable:end -->

anthonybilinski commented:
Looks good in my brief testing, but it would be good to get some real-world testing in different timing conditions to make sure nothing's out of order or gets stuck in weird states. This should probably be merged with at least a week or two of live testing before a release.

tox-user commented:
<img class="emoji" title=":lgtm_strong:" alt=":lgtm_strong:" align="absmiddle" src="https://reviewable.io/lgtm_strong.png" height="20" width="61"/>

---

Reviewed 6 of 6 files at r1.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/qtox/qtox/4975#-:-L7iotaXe_Y1I9ZEUWSs:b-jbsf8z)*
<!-- Sent from Reviewable.io -->

anthonybilinski committed:
fix(offlinemsg): make faux offline messages purely event based

Since real offline message reliability issue was fixed in PR #4607, now removing all the workarounds that had been added. Offline messages are now sent as soon as we see our friend come online, and at no other time. Fixes 2 minute wait time before attempting to send if message is entered while you or friend is offline, removes 2 minute constant retry timer, removes 250ms delay between seeing friend come online and sending offline messages.
